BlueprintUICommonControls/Sources/AttributedLabel.swift

```diff
@@ -43,53 +57,508 @@ public struct AttributedLabel: Element, Hashable {
+                    }
+
+                case .link:
+                    if let url = result.url {
+                        links.append(.init(text: url.absoluteString, range: result.range))
+                    }
+
+                case .address:
+                    if let addressComponents = result.addressComponents {
+                        let components = [
+                            addressComponents[.name],
+                            addressComponents[.organization],
+                            addressComponents[.street],
+                            addressComponents[.city],
+                            addressComponents[.zip],
+                            addressComponents[.country],
+                        ]
+
+                        let address = components
+                            .compactMap { $0 }
+                            .joined(separator: " ")
+
+                        if let urlQuery = address.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed),
+                           let url = URL(string: "https://maps.apple.com/?address=\(urlQuery)")
+                        {
+                            links.append(.init(text: url.absoluteString, range: result.range))
+                        }
+                    }
+
+                case .date:
+                    if let date = result.date,
+                       let url = URL(string: "calshow:\(date.timeIntervalSinceReferenceDate)")
+                    {
+                        links.append(.init(text: url.absoluteString, range: result.range))
+                    }
+
+                default:
+                    break
+                }
+            }
+
+            return links
+        }
+
+        private func accessibilityLinks(for links: [Link], in string: NSAttributedString) -> [LinkAccessibilityElement] {
+            links
+                .sorted(by: { $0.range.location < $1.range.location })
+                .compactMap { link in
+                    guard NSIntersectionRange(string.entireRange, link.range).length > 0 else {
+                        return nil
+                    }
+
+                    return LinkAccessibilityElement(
+                        container: self,
+                        label: string.attributedSubstring(from: link.range).string,
+                        link: link
+                    )
+                }
+        }
+
+        func applyLinkColors(activeLinks: [Link] = []) {
+            let mutableString = NSMutableAttributedString(attributedString: attributedText ?? .init(string: ""))
+
+            for link in links {
+                mutableString.addAttributes(linkAttributes, range: link.range)
+            }
+
+            for link in activeLinks {
+                mutableString.addAttributes(activeLinkAttributes, range: link.range)
+            }
+
+            attributedText = mutableString
+        }
+
+        override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
```

Comment:
I see  a`UIGestureRecognizerDelegate` conformance above – is that still needed given we're using these methods directly?

Reply:
- Remove `UIGestureRecognizerDelegate`